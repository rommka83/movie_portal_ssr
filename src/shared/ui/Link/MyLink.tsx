import React, { AnchorHTMLAttributes } from 'react';
import styles from './link.module.css';
import classNames from 'classnames';
import NextLink from 'next/link';

export const Link = React.memo<LinkProps>(({ className, to, children, ...props }) => {
  return (
    <NextLink {...props} className={classNames(styles.link, className)} href={to}>
      {children}
    </NextLink>
  );
});

type LinkProps = AnchorHTMLAttributes<HTMLAnchorElement> & {
  to: string;
};
