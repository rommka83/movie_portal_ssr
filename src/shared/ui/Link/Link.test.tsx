import renderer from 'react-test-renderer';
import { Link } from './MyLink';

describe('проверка Link', () => {
  test('отрисовка Link', () => {
    const component = renderer.create(<Link to='/CatalogPage/биография' />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
