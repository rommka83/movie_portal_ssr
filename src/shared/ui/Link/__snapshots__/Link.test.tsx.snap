// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка Link отрисовка Link 1`] = `
<a
  className="link"
  href="/CatalogPage/биография"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
/>
`;
